core_backend/tests/api/test_manage_ud_rules.py

```diff
@@ -0,0 +1,270 @@
+            headers={"Authorization": f"Bearer {readonly_token}"},
+        )
+        assert response.status_code == 200
+        assert response.json()["urgency_rule_text"] == urgency_rule_text
+        edited_metadata = response.json()["urgency_rule_metadata"]
+
+        assert all(edited_metadata[k] == v for k, v in urgency_rule_metadata.items())
+
+    async def test_edit_UDrules_not_found(
+        self, client: TestClient, fullaccess_token: str
+    ) -> None:
+        response = client.put(
+            "/urgency-rules/12345",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+            json={
+                "urgency_rule_text": "sample text",
+                "urgency_rule_metadata": {"key": "value"},
+            },
+        )
+
+        assert response.status_code == 404
+
+    async def test_list_UDrules(
+        self, client: TestClient, existing_rule_id: int, readonly_token: str
+    ) -> None:
+        response = client.get(
+            "/urgency-rules/", headers={"Authorization": f"Bearer {readonly_token}"}
+        )
+        assert response.status_code == 200
+        assert len(response.json()) > 0
+
+    async def test_delete_UDrules(
+        self, client: TestClient, existing_rule_id: int, fullaccess_token: str
+    ) -> None:
+        response = client.delete(
+            f"/urgency-rules/{existing_rule_id}",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+        )
+        assert response.status_code == 200
+
+
+class TestAuthManageContent:
+    @pytest.mark.parametrize(
+        "access_token, expected_status",
+        [("readonly_token", 400), ("fullaccess_token", 200)],
+    )
+    async def test_auth_delete(
+        self,
+        client: TestClient,
+        existing_rule_id: int,
+        access_token: str,
+        expected_status: int,
+        request: pytest.FixtureRequest,
+    ) -> None:
+        access_token = request.getfixturevalue(access_token)
+        response = client.delete(
+            f"/urgency-rules/{existing_rule_id}",
+            headers={"Authorization": f"Bearer {access_token}"},
+        )
+        assert response.status_code == expected_status
+
+    @pytest.mark.parametrize(
+        "access_token, expected_status",
+        [("readonly_token", 400), ("fullaccess_token", 200)],
+    )
+    async def test_auth_create(
+        self,
+        client: TestClient,
+        access_token: str,
+        expected_status: int,
+        request: pytest.FixtureRequest,
+    ) -> None:
+        access_token = request.getfixturevalue(access_token)
+        response = client.post(
+            "/urgency-rules",
+            headers={"Authorization": f"Bearer {access_token}"},
+            json={
+                "urgency_rule_text": "sample text",
+                "urgency_rule_metadata": {},
+            },
+        )
+        assert response.status_code == expected_status
+        if response.status_code == 200:
```

Comment:
Shouldn't we just have something (like a feature) to clean the db after every module ?

Reply:
Yeah probably. We can do test clean up in a different PR.